numpy/core/src/multiarray/getset.c

```diff
@@ -573,12 +583,12 @@ array_descr_set(PyArrayObject *self, PyObject *arg)
         Py_DECREF(temp);
     }
 
+    Py_DECREF(PyArray_DESCR(self));
```

Comment:
Is it safe to move this? It may have already have been overwritten. In any case, it is easier to be sure if it is left where it was.